### PR TITLE
Optimize code size by ising `?.` notation. NFC

### PR DESCRIPTION
**Pull request by sbc100**
And also using `!=` when we know the LHS are RHS are both string, specifically `typeof xx` on the left and a string literal on the right.

### CODE REVIEW

File: src/shell.js [hunk 1]
```diff
@@ -146,7 +146,7 @@ var ENVIRONMENT_IS_WASM_WORKER = Module['$ww'];
 #if SHARED_MEMORY && !MODULARIZE
 // In MODULARIZE mode _scriptDir needs to be captured already at the very top of the page immediately when the page is parsed, so it is generated there
 // before the page load. In non-MODULARIZE modes generate it here.
-var _scriptDir = (typeof document != 'undefined' && document.currentScript) ? document.currentScript.src : undefined;
+var _scriptDir = (typeof document != 'undefined') ? document.currentScript?.src : undefined;
```
Comment (curiousdannii) [inline src/shell.js:149]: Even shorter:
```suggestion
var _scriptDir = document?.currentScript?.src;
```

File: src/shell_minimal.js [hunk 2]
```diff
@@ -171,7 +171,7 @@ if (ENVIRONMENT_IS_NODE) {
 }
 #endif
 
-var currentScriptUrl = typeof _scriptDir != 'undefined' ? _scriptDir : ((typeof document != 'undefined' && document.currentScript) ? document.currentScript.src : undefined);
+var currentScriptUrl = typeof _scriptDir != 'undefined' ? _scriptDir : ((typeof document != 'undefined') ? document.currentScript?.src : undefined);
```
Comment (curiousdannii) [inline src/shell_minimal.js:174]:
```suggestion
var currentScriptUrl = _scriptDir ?? document?.currentScript?.src;
```

File: src/worker.js [hunk 3]
```diff
@@ -45,7 +45,7 @@ if (ENVIRONMENT_IS_NODE) {
       // __filename is undefined in ES6 modules, and import.meta.url only in ES6
       // modules.
 #if EXPORT_ES6
-      href: typeof __filename !== 'undefined' ? __filename : import.meta.url
+      href: typeof __filename != 'undefined' ? __filename : import.meta.url
```
Comment (curiousdannii) [inline src/worker.js:48]:
```suggestion
      href: __filename ?? import.meta.url
```

**Review by curiousdannii — CHANGES_REQUESTED** [review]
More please! 🙂

File: src/shell.js (hunk 1 above)
Comment (sbc100) [reply src/shell.js:149]: I don't think `foo?` work for global sadly.  i.e. you can't do `foo?.method()` but you can do `globalThis.foo?.method()`.

Maybe if we knew we have univeral access to globalThis we should use that here, but I'm not sure we can assume that yet

Comment (curiousdannii) [reply src/shell.js:149]: Huh. I thought it would work, but didn't test sorry.

**Review by kripken — APPROVED** [review]

**Comment by kripken** [comment]
`ising` in title should be `using`

**Pull request merged**